Subject: [PATCH] remove warning filter
---
Index: src/ada/core/utils.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ada/core/utils.py b/src/ada/core/utils.py
--- a/src/ada/core/utils.py	(revision d83292ff31e93d75c8c77c2bf29a7dcd24631561)
+++ b/src/ada/core/utils.py	(date 1734101724868)
@@ -73,9 +73,6 @@
 
 def roundoff(x: float, precision=Config().general_precision) -> float:
     """Round using a specific number precision using the Decimal package"""
-    import warnings
-
-    warnings.filterwarnings(action="error", category=np.ComplexWarning)
     xout = float(Decimal(float(x)).quantize(Decimal("." + precision * "0" + "1"), rounding=ROUND_HALF_EVEN))
     return xout if abs(xout) != 0.0 else 0.0
 
